ee/hogai/eval/ci/eval_memory_onboarding.py

```diff
@@ -0,0 +1,397 @@
+from typing import Optional
+from uuid import uuid4
+
+import pytest
+
+from autoevals.llm import LLMClassifier
+from braintrust import EvalCase
+from braintrust_core import Score
+
+from posthog.schema import AssistantMessage, EventTaxonomyItem, HumanMessage
+
+from posthog.models.user import User
+from posthog.sync import database_sync_to_async
+
+from ee.hogai.graph import AssistantGraph
+from ee.hogai.graph.memory.prompts import ENQUIRY_INITIAL_MESSAGE, SCRAPING_TERMINATION_MESSAGE
+from ee.hogai.graph.root.nodes import SLASH_COMMAND_INIT
+from ee.hogai.utils.types import AssistantNodeName, AssistantState
+from ee.models.assistant import Conversation
+
+from ..base import MaxPublicEval
+
+
+class MemoryLLMClassifier(LLMClassifier):
+    def _run_eval_sync(self, output, expected, **kwargs):
+        if not output:
+            return Score(name=self._name(), score=None)
+        if expected in [ENQUIRY_INITIAL_MESSAGE, SCRAPING_TERMINATION_MESSAGE]:
+            # For the special cases, we MUST see the expected messages verbatim
+            return Score(
+                name=self._name(),
+                score=1.0 if isinstance(output, AssistantMessage) and output.content == expected else 0.0,
+            )
+        return super()._run_eval_sync(output, expected, **kwargs)
+
+    async def _run_eval_async(self, output, expected, **kwargs):
+        if not output:
+            return Score(name=self._name(), score=None)
+        if expected in [ENQUIRY_INITIAL_MESSAGE, SCRAPING_TERMINATION_MESSAGE]:
+            # For the special cases, we MUST see the expected messages verbatim
+            return Score(
+                name=self._name(),
+                score=1.0 if isinstance(output, AssistantMessage) and output.content == expected else 0.0,
+            )
+        return await super()._run_eval_async(output, expected, **kwargs)
+
+
+class SatisfiesProductDetails(MemoryLLMClassifier):
+    """Binary check: Does the scraped content contain identifiable product features?"""
+
+    def __init__(self, **kwargs):
+        super().__init__(
+            name="satisfies_product_details",
+            prompt_template="""Determine if the scraped content contains identifiable product features.
+
+A PASS requires:
+- At least 2-3 specific product features are mentioned
+- Features are described with actual names or functionality (not generic)
+- Features relate to the actual product being analyzed
+
+A FAIL means:
+- No specific features mentioned, only vague descriptions
+- Features are generic or unrelated to the actual product
+- Content is purely marketing speak without concrete functionality
+
+Input: {{input}}
+Output: {{output}}
+
+Does the output contain identifiable product features? Be brutal.
+- pass: Contains specific, identifiable product features
+- fail: No identifiable product features found""",
+            choice_scores={
+                "pass": 1.0,
+                "fail": 0.0,
+            },
+            model="gpt-4.1",
```

Comment:
Not a bit fan of defining models as strings (easy to miss later if we decide to switch), but we had this discussion before 😄